contracts/utils/convex/ConvexStakingWrapper.sol

```diff
@@ -0,0 +1,452 @@
+            reward.reward_remaining = 0;
+        }
+    }
+
+    function rewardLength() external view returns (uint256) {
+        return rewards.length;
+    }
+
+    function _getDepositedBalance(address _account) internal view virtual returns (uint256) {
+        if (_account == address(0) || _account == collateralVault) {
+            return 0;
+        }
+        //get balance from collateralVault
+
+        return _balanceOf[_account];
+    }
+
+    function _getTotalSupply() internal view virtual returns (uint256) {
+        return _totalSupply;
+    }
+
+    function _calcCvxIntegral(
+        address[2] memory _accounts,
+        uint256[2] memory _balances,
+        uint256 _supply,
+        bool _isClaim
+    ) internal {
+        uint256 bal = IERC20(cvx).balanceOf(address(this));
+        uint256 d_cvxreward = bal - cvx_reward_remaining;
+
+        if (_supply > 0 && d_cvxreward > 0) {
+            cvx_reward_integral = cvx_reward_integral + (d_cvxreward * 1e20) / (_supply);
+        }
+
+        //update user integrals for cvx
+        for (uint256 u = 0; u < _accounts.length; u++) {
+            //do not give rewards to address 0
+            if (_accounts[u] == address(0)) continue;
+            if (_accounts[u] == collateralVault) continue;
+
+            uint256 userI = cvx_reward_integral_for[_accounts[u]];
+            if (_isClaim || userI < cvx_reward_integral) {
+                uint256 receiveable = cvx_claimable_reward[_accounts[u]] +
+                    ((_balances[u] * (cvx_reward_integral - userI)) / 1e20);
+                if (_isClaim) {
+                    if (receiveable > 0) {
+                        cvx_claimable_reward[_accounts[u]] = 0;
+                        IERC20(cvx).safeTransfer(_accounts[u], receiveable);
+                        bal = bal - (receiveable);
+                    }
+                } else {
+                    cvx_claimable_reward[_accounts[u]] = receiveable;
+                }
+                cvx_reward_integral_for[_accounts[u]] = cvx_reward_integral;
+            }
+        }
+
+        //update reward total
+        if (bal != cvx_reward_remaining) {
+            cvx_reward_remaining = bal;
+        }
+    }
+
+    function _calcRewardIntegral(
+        uint256 _index,
+        address[2] memory _accounts,
+        uint256[2] memory _balances,
+        uint256 _supply,
+        bool _isClaim
+    ) internal {
+        RewardType storage reward = rewards[_index];
+
+        //get difference in balance and remaining rewards
+        //getReward is unguarded so we use reward_remaining to keep track of how much was actually claimed
+        uint256 bal = IERC20(reward.reward_token).balanceOf(address(this));
+        // uint256 d_reward = bal-(reward.reward_remaining);
+
+        if (_supply > 0 && bal - (reward.reward_remaining) > 0) {
+            reward.reward_integral = reward.reward_integral + uint128(bal - (reward.reward_remaining * 1e20) / _supply);
```

Comment:
the original code is:
```
reward.reward_integral = reward.reward_integral + uint128(bal.sub(reward.reward_remaining).mul(1e20).div(_supply));
```
Which is: `reward_integral = reward_integral + (bal - reward_remaining) * 1e20 / _supply`
You're doing `reward_integral = reward_integral + (bal - reward_remaining * 1e20 / _supply)`